be/src/storage/rowset/vectorized/segment_iterator.cpp

```diff
@@ -647,6 +673,7 @@ Status SegmentIterator::_do_get_next(Chunk* result, vector<rowid_t>* rowid) {
     _context->_read_chunk->reset();
     _context->_dict_chunk->reset();
     _context->_final_chunk->reset();
```

Comment:
final chunk means last chunk, handle `_adapt_chunk` after `_final_chunk`, which seems not reasonable.